providers/os/resources/docker_file_test.go

```diff
@@ -12,70 +12,197 @@ import (
 )
 
 func TestParseDockerfile(t *testing.T) {
-	dockerfile := `
+	cases := []struct {
+		purpose           string
+		subjectDockerFile string
+
+		expectedLabels          map[string]interface{}
+		expectedEnv             map[string]interface{}
+		expectedFromImage       string
+		expectedFromTag         string
+		expectedUser            plugin.TValue[*mqlDockerFileUser]
+		expectedCmd             plugin.TValue[*mqlDockerFileRun]
+		expectedEntrypoint      plugin.TValue[*mqlDockerFileRun]
+		expectedRunStruct       []plugin.TValue[*mqlDockerFileRun]
+		expectedCopyStruct      []plugin.TValue[*mqlDockerFileCopy]
+		expectedAddStruct       []plugin.TValue[*mqlDockerFileAdd]
+		expectedExposeStructArr []plugin.TValue[*mqlDockerFileExpose]
+	}{
+		{
+			purpose: "minimal instructions with CMD",
+			subjectDockerFile: `
+FROM alpine
+CMD ["/bin/sh", "-c", "echo 'Hola'"]
+`,
+			expectedLabels:    map[string]interface{}{},
+			expectedEnv:       map[string]interface{}{},
+			expectedFromImage: "alpine",
+			expectedCmd: plugin.TValue[*mqlDockerFileRun]{
+				Data: &mqlDockerFileRun{
+					Script: plugin.TValue[string]{Data: "/bin/sh\n-c\necho 'Hola'"},
+				},
+			},
+		},
```

Comment:
@scottford-io feel free to add more test cases like this one, you have some useful ones in your description.

<img src="https://media3.giphy.com/media/B0vFTrb0ZGDf2/giphy.gif"/>

Reply:
Thank you @afiune 👍